# Transcript assembly improves expression quantification of transposable elements in single-cell RNA-seq data #

[The paper](https://genome.cshlp.org/content/31/1/88.short) is in Genome Research (2021)

The author is [Wanqing Shao](https://scholar.google.co.uk/citations?hl=en&user=cQFVP10AAAAJ&view_op=list_works&sortby=pubdate) (a postdoc at the [Wang Lab](http://wang.wustl.edu/research))


- [Introduction](#Introduction)
- Results: Technical aspects
  - A higher percentage of reads are mapped to TEs in scRNA-seq compared with bulk RNA-seq 
  - Counting scRNA-seq reads at individual TEs leads to large numbers of false positive candidates
- Results: Their proposal
  - Transcript assembly improves TE expression analysis
- Results: Applying their proposal to biology
  - Dynamic TE expression in preimplantation embryos
  - Tissue-specific TE expression during mouse gastrulation and early organogenesis
- Discussion
- Methods
  - Data processing of bulk RNA-seq data sets
  - Data processing of scRNA-seq data sets
  - Constructing TE transcripts
  - Read assignments
  - RNA-seq simulation
  - Mouse early embryogenesis scRNA-seq data set analysis
  - TE transcript clustering in mouse preimplantation stages
  - TE subfamily/family enrichment analysis
  - Other statistical analysis and figure generation
  - Publicly available data sets used in this study
  - Software availability


## Introduction ##

- Lots of TE-derived sequences can't transpose anymore, but they are transcribed into TE-RNA chimeric transcripts and/or noncoding RNAs.
- Most (2/3) mature long ncRNAs contain at least one TE.
- specific TE-ncRNA transcripts are present at specific development stages and tissues.
- knocking out specific TE families leads to development defects.
- To quantify the expression of repetitive elements in bulk RNA-seq data:
  - tools aggregate multialigned reads in subfamilies/families
  - tools redistribute multialigned reads to individual TEs based on heuristic/statistical rules

In single cell data:

- Counting reads at individual TEs or subfamilies/families could be wrong because:
  - One full transcript can contain multiple TEs from different subfamilies/families.
  - So, different expression values might be assigned to individual TEs that are actually within the same transcript.

- Their solution is to assemble transcripts from bulk RNA-seq data and use them to quantify TE expression in single-cell data


## Results ##

### Technical aspects ###


